cmd/minishift/cmd/root.go

```diff
@@ -80,27 +83,62 @@ var RootCmd = &cobra.Command{
 			isFreshStart bool
 		)
 
-		if !filehelper.Exists(constants.Minipath) || filehelper.IsEmptyDir(constants.Minipath) {
+		//We need to initialize allinstance config as the profile information would be there
```

Comment:
I think this method needs really to be split up. This will help to make clear what's actually happening as well.